### PR TITLE
Shirinhasan/ravhibq 672/handling invalid escape sequences

### PR DESCRIPTION
**Pull request by shirin-hasan**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 1]
```diff
@@ -308,6 +308,15 @@ public BigQuerySqlDialect(SqlDialect.Context context) {
   private static final String SHIFTLEFT = "<<";
   private static final String BITNOT = "~";
 
+  public static final Map<String, String> BQ_VALID_ESCAPE_SEQUENCES =
```
Comment (divyanshuS27) [inline core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:311]: Rename to 'STRING_LITERAL_ESCAPE_SEQUENCES'

Comment (shirin-hasan) [reply core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java]: Done

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 2]
```diff
@@ -1459,12 +1462,56 @@ private void unparseRegexMatchCount(SqlWriter writer, SqlCall call,
...
+  private void unparseRegexpReplace(SqlWriter writer, SqlCall call,
+      int leftPrec, int rightPrec) {
+    int indexOfRegexOperand = 1;
+    SqlWriter.Frame regexpReplaceFrame = writer.startFunCall("REGEXP_REPLACE");
+    List<SqlNode> operandList = call.getOperandList();
+    unparseRegexFunctionsOperands(writer, leftPrec, rightPrec, indexOfRegexOperand, operandList);
+    writer.endFunCall(regexpReplaceFrame);
+  }
+
+  private void unparseRegexpInstr(SqlWriter writer, SqlCall call,
```
Comment (divyanshuS27) [inline core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:1483]: possibly the redundant code could have been made a single function with additional argument.

Comment (shirin-hasan) [reply core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:1483]: As discussed, we'll address this in another MR.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 3]
```diff
@@ -11141,7 +11141,7 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
         + "WHERE \"first_name\" ='Maya\n Gutierrez'";
     final String expectedBQSql = "SELECT *\n"
         + "FROM foodmart.employee\n"
-        + "WHERE first_name = 'Maya Gutierrez'";
```
[the remark this answers is not in the document]
Comment (shirin-hasan) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:11144]: Will discuss and inform them about the changes

**Review by dm-shantanuc — APPROVED** [review]
Was reviewed by Divyanshu.

**Pull request merged**